src/java/org/apache/cassandra/db/filter/SliceQueryFilter.java

```diff
@@ -273,6 +275,7 @@ public void collectReducedColumns(ColumnFamily container, Iterator<Cell> reduced
     {
         reducedCells = CountingCellIterator.wrapIterator(reducedColumns, now, gcBefore);
         columnCounter = columnCounter(container.getComparator(), now);
+        rangeTombstoneCounter = RangeTombstoneCounter.create(gcBefore, container.deletionInfo());
```

Comment:
Note -- we cannot use cell counting iterator as ranged tombstones are not treated as cells (and our filtered out prematurely). I thought this would be the least invasive with minimal perf degradation that we could do.